src/common/lock_manager.h

```diff
@@ -81,6 +81,8 @@ class LockManager {
 
 class LockGuard {
  public:
+  LockGuard() = default;
```

Comment:
This ctor will make the state of a LockGuard object more complicated, which is not what we want in my view.

You can refer to https://en.cppreference.com/w/cpp/thread/lock_guard/lock_guard that also doesn't provide such a ctor.

And if you do not want to initialize the object when you define it, please use `std::optional`